### PR TITLE
CF log bucket and kms key encryption

### PR DESCRIPTION
**Pull request by willwatters**

### CODE REVIEW

File: groups/promo-infrastructure/locals.tf [hunk 1]
```diff
@@ -6,7 +6,29 @@ locals {
...
+  security_s3                      = data.vault_generic_secret.security_s3.data
+
+  kms_customer_master_keys = {
+    ssm = {
+      description             = "Session Manager data encryption key"
+      deletion_window_in_days = 30
+      enable_key_rotation     = true
+      is_enabled              = true
+      key_usage_foreign_account_ids = [
+        local.account_ids[var.aws_account]
```
Comment (mattystone) [inline groups/promo-infrastructure/locals.tf:19]: Unneeded? This is adding the local account into the key policy, but that already exists by default. This is only required to add cross account permissions for KMS keys.

File: groups/promo-infrastructure/locals.tf [hunk 2]
```diff
@@ -6,7 +6,29 @@ locals {
...
+      ]
+    },
+    promo = {
+      description             = "Promo encryption key"
+      deletion_window_in_days = 30
+      enable_key_rotation     = true
+      is_enabled              = true
+      key_usage_foreign_account_ids = [
+        local.account_ids[var.aws_account]
```
Comment (mattystone) [inline groups/promo-infrastructure/locals.tf:28]: Same as above, not required.

File: groups/promo-infrastructure/locals.tf (hunk 1 above)
Comment (willwatters) [reply groups/promo-infrastructure/locals.tf:19]: This has been removed

File: groups/promo-infrastructure/locals.tf (hunk 2 above)
Comment (willwatters) [reply groups/promo-infrastructure/locals.tf:28]: Same as above

**Pull request merged**